pulsar-client-api/src/main/java/org/apache/pulsar/client/api/Consumer.java

```diff
@@ -390,6 +390,21 @@ void reconsumeLater(Message<?> message,
      */
     void acknowledgeCumulative(MessageId messageId) throws PulsarClientException;
 
+    /**
+     * Acknowledge the reception of all the messages in the stream up to (and including) the provided message for each topic or partition.
+     *
+     * <p>This method will block until the acknowledge has been sent to the brokers of each topic or partition.
```

Comment:
This is not correct.
Please remove this sentence at all.
Acks are always sent best effort.
Only of you do something in a transaction you can have such guarantees.